src/components/config.cpp

```diff
@@ -101,16 +101,18 @@ void config::parse_file() {
     }
 
     if (key == "include-file") {
-      auto file_path = file_util::expand(value);
+      auto file_path = file_util::is_absolute(value)
```

Comment:
Can you put this into a seperate function that takes any path and returns an expanded absolute path please.

Reply:
Good call. I added `expand_relative_to` as a function.

I wasn't sure the best order of the parameters: putting the directory first could make sense, since that's the order they would be appended together (`dir + '/' + path`). I put the path to be expanded first, since that kept the ordering consistent with normal `expand`.

I also don't call `dirname` on the `base_dir` argument. There's no way to inspect a path to decide if it's a file or a directory, and calling `stat` on the path seemed like overkill. Always calling dirname wouldn't work, since if someone had a directory and not a file argument, they'd need to add a fake file name component to get the result they want.

I don't call expand on the base_dir either, since it's likely already been done (plus, the caller can do that if they want to).

I don't assert that `base_dir` is an absolute path, because I could imagine a scenario where you want to expand a file relative to an absolute path.